### ISSUE TITLE
[Feature Request]: Sort By Video Length

### ISSUE THREAD

**Issue by whyyn0tt** [yuliskov/SmartTube#3921]
### Describe the feature you'd like to request

I'd love to have the option to sort a playlist by video length, shortest to longest. 

### Describe the solution you'd like

An additional option under the Sort Playlist menu would be ideal. 

### Describe alternatives you've considered

I currently use a Greasemonkey script on my PC to do this via web, but would love to see similar implemented in this app.

**Comment by dr04e606** [comment]
Yes, this feature would be very useful, especially for longer playlists...

**Comment by ScottFries** [comment]
Got to looking into this, but they're just calling back to YouTube's API to do the sort, so it's currently limited to only the sort options YouTube provides (which they've implemented all of). If YouTube added the option to sort by length then this change is trivial:

- Adding 2 new strings after common/src/main/res/values/strings.xml:458 and their subsequent localizations
- Adding 2 new ints to MediaItemService in MediaServiceCore/mediaserviceinterfaces/src/main/java/com/liskovsoft/mediaserviceinterfaces/MediaItemService.java:21
- Adding the 2 new sort options at common/src/main/java/com/liskovsoft/smartyoutubetv2/common/utils/AppDialogUtil.java:1091
